src/App.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import { Provider } from 'react-redux';
+
+import MainPage from './components/MainPage/MainPage.jsx';
+import ErrorPage from './components/ErrorPage/ErrorPage.jsx';
+import NotFoundPage from './components/NotFoundPage/NotFoundPage.jsx';
+import BalancePage from './components/BalancePage/BalancePage.jsx';
+
+import store from './redux/store.js';
+
+const App = () => (
+  <Provider store={store}>
+    <Router>
+      <Switch>
+        <Route exact path="/">
+          <MainPage />
+        </Route>
+        <Route path="/balance">
+          <BalancePage />
+        </Route>
+        <Route path="/error">
+          <ErrorPage />
+        </Route>
+        <Route path="*">
```

Comment:
probably you can even leave here 
```
<Route>
   <NotFoundPage />
</Route>
```

_________
basically, the structure of the router here is strange
I know it works,but I haven't seen the structure inside my work projects, so I can advise to use the following
```
<Route path={ROUTES.UPDATE_DIAGNOSIS} component={UpdateDiagnosis} />
```

Reply:
Fixed. Routes to components are now attribute based;